packages/abstraction/test/ContractGenerator.test.ts

```diff
@@ -1,3 +1,98 @@
+import { ContractGenerator } from "../src";
+import path from 'path';
+import { MultipleContractAdapter, EOSContractAdapter, ICONContractAdapter, ModelContractAdapter } from "@aloxide/bridge";
+import * as  SchemaValidator from '../src/SchemaValidator';
+
 describe('test ContractGenerator', () => {
-  it.todo('test contract generator');
+  let contractGenerator;
+  const logger = {
+    info: jest.fn(),
+    debug: jest.fn(),
+  };
+  const adapter = new MultipleContractAdapter();
+
+  adapter.addAdapters(
+    new EOSContractAdapter(),
+    new ICONContractAdapter(),
+    new ModelContractAdapter(),
+  );
+
+  const checkName = val => {
+    return /^[1-5a-zA-Z]+/.test(val)
+  }
+
+  const checkType = val => {
+    const supportedType = ["uint64_t", "number", "string", "array", "bool"];
+    return supportedType.indexOf(val) !== -1
+  }
+
+  const requiredSchema = {
+    entities: [{
+      name: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } },
+      fields: [{
+        name: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } },
+        type: { type: String, required: true, use: { checkType } },
+      }],
+      key: { type: String, required: true, length: { min: 1, max: 12 }, use: { checkName } }
+    }
+    ]
+  }
+
+  beforeEach(() => {
+    jest.restoreAllMocks();
+    contractGenerator = new ContractGenerator({
+      aloxideConfigPath: path.resolve(__dirname, './aloxide.yml'),
+      resultPath: path.resolve(__dirname, './out'),
+      adapter,
+      logger
+    });
+  })
+
+  describe('test validateEntity function', () => {
+    it('test validate entity successfully', async () => {
+
+      const validateSchemaMock = jest.spyOn(SchemaValidator, 'validateSchema').mockReturnValueOnce([]);
+      const result = await contractGenerator.validateEntity();
+      expect(result).toBeTruthy();
+      expect(validateSchemaMock).toHaveBeenCalledTimes(1);
+      expect(validateSchemaMock.mock.calls[0][0]).toBe(contractGenerator.aloxideConfig);
```

Comment:
You might use less code with `toBeCalledWith` here.

Reply:
I try but it doesn't work with this case

Reply:
How did it not work? it's a documented API. Can you provide error here?

Reply:
![Screen Shot 2020-09-17 at 16 01 14](https://user-images.githubusercontent.com/38729663/93449648-30f60580-f8ff-11ea-8808-72cd2219f128.png)

Reply:
Okay, I can't find out why the result is so strange like that too. There must be something wrong with Jest or our code, but let's keep your working version for now.